src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNOWindow.m

```diff
@@ -639,6 +656,95 @@ void uno_set_window_close_callbacks(window_should_close_fn_ptr shouldClose, wind
     return gr_direct_context_make_metal(device, queue);
 }
 
+CGFloat readNextCoord(const char *svg, int *position, long length)
+{
+    CGFloat result = NAN;
+    if (*position >= length) {
+#if DEBUG
+        NSLog(@"uno_window_clip_svg readNextCoord position:%d >= length:%ld", *position, length);
+#endif
+        return result;
+    }
+    const char* start = svg + *position;
+    char* end;
+    result = strtod(start, &end);
+    *position += (int)(end - start);
+    return result;
+}
+
+void uno_window_clip_svg(UNOWindow* window, const char* svg)
+{
+    if (svg) {
+#if DEBUG
+        NSLog(@"uno_window_clip_svg %@ %@ %s", window, window.contentView.layer.description, svg);
+#endif
+        NSArray<__kindof NSView *> *subviews = window.contentViewController.view.subviews;
+        for (int i = 0; i < subviews.count; i++) {
+            NSView* view = subviews[i];
+#if DEBUG
+            NSLog(@"uno_window_clip_svg subview %d %@ layer %@ mask %@", i, view, view.layer, view.layer.mask);
+#endif
+            CGMutablePathRef path = CGPathCreateMutable();
```

Comment:
Would it be of interest to cache those instances based on the string hash?

Reply:
Worth measuring. There's a nice [`NSCache`](https://developer.apple.com/documentation/foundation/nscache?language=objc) available (so we don't have to do the hashing ourselves).